cmp/leftrightchooser/LeftRightChooserFilter.js

```diff
@@ -49,17 +51,21 @@ class LeftRightChooserFilter extends Component {
     }
     
     runFilter() {
-        const fields = this.props.fields || [],
+        const {fields = [], anyMatch} = this.props,
             searchTerm = escapeRegExp(this.value);
 
         const filter = (raw) => {
             return fields.some(f => {
                 const fieldVal = !!searchTerm && raw[f];
-                return ((fieldVal && new RegExp(`(^|\\\\W)${searchTerm}`, 'ig').test(fieldVal)) || !fieldVal);
+                return ((fieldVal && new RegExp(`${anyMatch ? '' : '(^|\\\\W)'}${searchTerm}`, 'ig').test(fieldVal)) || !fieldVal);
```

Comment:
I would suggest we break this up into a few lines. It's pretty long, and with a variable-driven regex in the midst of it it's a lot to grok!